src/CADL.Extension/Emitter.Csharp/src/type/InputType.ts

```diff
@@ -20,11 +20,19 @@ export interface InputLiteralType extends InputType {
     Value: any;
 }
 
+export function isInputLiteralType(type: InputType): type is InputLiteralType {
+    return type.Name === "Literal";
+}
+
 export interface InputUnionType extends InputType {
     Name: "Union";
     UnionItemTypes: InputType[];
 }
 
+export function isInputUnionType(type: InputType): type is InputUnionType {
```

Comment:
Can we not check the name, instead to check the core property `UnionItemTypes`?

e..g if ( (type as InputUnionType).UnionItemTypes !== undefined)

Reply:
See the definition of `InputUnionType` above, it always has the name of "Union". So it is the best to check by name. Another reason is: if we use the cast in the body, we lose the meaning of adding this function whose return type is type predicate. I add this function to avoid casting.